ReminderApp-OBJC/ReminderApp-OBJC/Classes/Controller/CDReminderTableViewController.m

```diff
@@ -129,62 +88,229 @@ - (void)controller:(NSFetchedResultsController *)controller didChangeObject:(id)
 			CDReminder *reminderItem = [self.fetchedResultsController objectAtIndexPath:indexPath];
 			cell.textLabel.text = reminderItem.taskName;
-			cell.detailTextLabel.text = reminderItem.taskHoure;
 		}
 			break;
 		case NSFetchedResultsChangeMove:
-			[self.tableView deleteRowsAtIndexPaths:[NSArray arrayWithObject:indexPath]
+			[self.tableView deleteRowsAtIndexPaths:@[indexPath]
 								  withRowAnimation:UITableViewRowAnimationFade];
-			[self.tableView insertRowsAtIndexPaths:[NSArray arrayWithObject:newIndexPath]
+			[self.tableView insertRowsAtIndexPaths:@[newIndexPath]
 								  withRowAnimation:UITableViewRowAnimationFade];
 			break;
-		default:
-			break;
 	}
 	
 }
 
-- (void) controllerDidChangeContent:(NSFetchedResultsController *)controller {
+- (void)controllerDidChangeContent:(NSFetchedResultsController *)controller {
 	
 	[self.tableView endUpdates];
 	
 }
 
-#pragma mark - DataSource and Delegate
+#pragma mark - Helper methods
+
+- (CGFloat)textViewHeightForRowAtIndexPath:(NSIndexPath *)indexPath {
 
-- (NSInteger)numberOfSectionsInTableView:(UITableView *) tableView {
+	UITextView *calculationView = ((CDAddReminderCell *)[self.tableView cellForRowAtIndexPath:indexPath]).textView;
+	CGFloat textViewWidth = calculationView.frame.size.width;
 	
-	return self.fetchedResultsController.sections.count;
+	if (!calculationView) {
+		textViewWidth = 290;
+		return 50;
+	}
 
+	CGSize size = [calculationView sizeThatFits:CGSizeMake(textViewWidth, FLT_MAX)];
+	return size.height;
+	
 }
 
-- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
+- (void)scrollToCursorForTextView:(UITextView *)textView {
+	
+	CGRect cursorRect = [textView caretRectForPosition:textView.selectedTextRange.start];
+	cursorRect = [self.tableView convertRect:cursorRect fromView:textView];
+	
+	if (![self rectVisible:cursorRect]) {
+		cursorRect.size.height += 8;
+		[self.tableView scrollRectToVisible:cursorRect animated:YES];
+	}
+	
+}
 
-	return self.fetchedResultsController.sections[section].numberOfObjects;
+- (BOOL)rectVisible:(CGRect)rect {
+	
+	CGRect visibleRect;
+	visibleRect.origin = self.tableView.contentOffset;
+	visibleRect.origin.y += self.tableView.contentInset.top;
+	visibleRect.size = self.tableView.bounds.size;
+	visibleRect.size.height -= self.tableView.contentInset.top + self.tableView.contentInset.bottom;
+	
+	return CGRectContainsRect(visibleRect, rect);
+	
+}
 
+- (void)keyboardWillShow:(NSNotification *)aNotification {
+	
+	NSDictionary* info = [aNotification userInfo];
+	CGSize kbSize = [[info objectForKey:UIKeyboardFrameBeginUserInfoKey] CGRectValue].size;
+	
+	UIEdgeInsets contentInsets = UIEdgeInsetsMake(self.tableView.contentInset.top, 0.0, kbSize.height, 0.0);
+	
+	self.tableView.contentInset = contentInsets;
+	self.tableView.scrollIndicatorInsets = contentInsets;
+	
+}
+
+- (void)keyboardWillHide:(NSNotification*)aNotification {
+	
+	[UIView beginAnimations:nil context:nil];
```

Comment:
Varianta asta de cod e invechita. In documentatie scrie "Use of this method is discouraged in iOS 4.0 and later. You should use the block-based animation methods to specify your animations instead." Cauta un exemplu cu varianta pe block sau adapteaza tu codul pe nou a metoda.